lib/sycamore/sycamore/transforms/merge_elements.py

```diff
@@ -614,6 +614,197 @@ def postprocess_element(self, elem: Element) -> Element:
+            else:
+                if not first_header:
+                    element.data["_header"] = None
+
+        to_merge = [self.preprocess_element(e) for e in document.elements]
+        new_elements = [to_merge[0]]
+        for element in to_merge[1:]:
+            if self.should_merge(new_elements[-1], element):
+                new_elements[-1] = self.merge(new_elements[-1], element)
+            else:
+                new_elements.append(element)
+        document.elements = [
+            self.postprocess_element(e) for e in new_elements if e.type not in ["Section-header", "Title"]
+        ]
+        return document
+
+    def should_merge(self, element1: Element, element2: Element) -> bool:
+        # deal with empty elements
+        if (
+            DocumentPropertyTypes.PAGE_NUMBER not in element1.properties
+            or DocumentPropertyTypes.PAGE_NUMBER not in element2.properties
+            or element1.type is None
+            or element2.type is None
+        ):
+            return False
+
+        # Conditionally prevent merging across pages
+        if (
+            not self.merge_across_pages
+            and element1.properties[DocumentPropertyTypes.PAGE_NUMBER]
+            != element2.properties[DocumentPropertyTypes.PAGE_NUMBER]
+        ):
+            return False
+
+        if element1.data["token_count"] + 1 + element2.data["token_count"] > self.max_tokens and element2.type not in [
+            "Section-header",
+            "Title",
+        ]:
+            # Add header to next element
+            element2["_header"] = element1["_header"]
+            if element1.data["_header"]:
+                if element2.text_representation:
+                    element2.text_representation = element1.data["_header"] + "\n" + element2.text_representation
+                else:
+                    element2.text_representation = element1.data["_header"]
+            return False
+
+        # Merge consecutive section headers/titles and save as a section-header element
+        if element1.type in ["Section-header", "Title"] and element2.type in ["Section-header", "Title"]:
+            return True
+
+        # MERGE adjacent 'text' elements
+        small = {"Text", "List-item", "Caption", "Footnote", "Formula", "Page-footer", "Page-header"}
+        if (element1 is not None) and (element1.type in small) and (element2 is not None) and (element2.type in small):
+            return True
+
+        # Add header to next element (images, tables)
+        if element2.type not in ["Section-header", "Title"]:
+            element2.data["_header"] = element1.data["_header"]
+            if element2.text_representation:
+                if element2.data["_header"]:
+                    element2.text_representation = element2.data["_header"] + "\n" + element2.text_representation
+            else:
+                element2.text_representation = element2.data["_header"]
+        return False
```

Comment:
Do we want to rewrite the end to error on unknown element types or do we want to keep it as-is so that people can chain mergers (`GreedySectionMerger` creates elements of type `"Section-header+table"`)?

Reply:
I think I will add "Section" in the list of elements allowed to be merged(from greedytextelement merger). For your comment, let us keep as is, since augment header is only going to make the chunking better?

Reply:
Ok.